### PR TITLE
Refactored the RegexFunctionVisitor and expanded test cases

### PR DESCRIPTION
**Pull request by apmoriarty**
Addresses https://github.com/NationalSecurityAgency/datawave/issues/1574

### CODE REVIEW

File: warehouse/query-core/src/main/java/datawave/query/jexl/visitors/RegexFunctionVisitor.java [hunk 1]
```diff
@@ -47,61 +50,56 @@ public Object visit(ASTFunctionNode node, Object data) {
...
     private JexlNode buildRegexNode(ASTIdentifier identifier, String functionName, String regex) {
         String field = JexlASTHelper.deconstructIdentifier(identifier.image);
         if (nonEventFields.contains(field.toUpperCase())) {
             try {
                 JavaRegexAnalyzer jra = new JavaRegexAnalyzer(regex);
                 if (!jra.isNgram()) {
-                    if (functionName.equals("includeRegex")) {
-                        if (log.isDebugEnabled())
-                            log.debug("Building new ER Node");
+                    if (functionName.equals(INCLUDE_REGEX)) {
```
Comment (avgAGB) [inline warehouse/query-core/src/main/java/datawave/query/jexl/visitors/RegexFunctionVisitor.java:119]: Did this visitor end up being a bit too chatty with debug messages when the ER node was built? Was wondering if there is anything here that would be considered loggable even at a trace level.

Comment (apmoriarty) [reply warehouse/query-core/src/main/java/datawave/query/jexl/visitors/RegexFunctionVisitor.java:119]: The debug message wasn't informative so I decided to just rip out all logging for this visitor. If you would like I can add in a trace level message that logs when a filter function is rewritten

Comment (avgAGB) [reply warehouse/query-core/src/main/java/datawave/query/jexl/visitors/RegexFunctionVisitor.java:119]: That sounds good to me, thanks.

Comment (avgAGB) [reply warehouse/query-core/src/main/java/datawave/query/jexl/visitors/RegexFunctionVisitor.java:104]: That sounds good to me, thanks.

Comment (apmoriarty) [reply warehouse/query-core/src/main/java/datawave/query/jexl/visitors/RegexFunctionVisitor.java:119]: Added the logger back and put in a more informative message

**Review by avgAGB — DISMISSED** [review]

File: warehouse/query-core/src/main/java/datawave/query/jexl/visitors/RegexFunctionVisitor.java [hunk 2]
```diff
@@ -47,61 +56,66 @@ public Object visit(ASTFunctionNode node, Object data) {
...
+                            log.trace("Rewrote \"" + JexlStringBuildingVisitor.buildQueryWithoutParse(node) + "\" into \""
+                                            + JexlStringBuildingVisitor.buildQueryWithoutParse(children.get(0)) + "\"");
                         }
-                    }
-                }
-                if (newParent.jjtGetNumChildren() == node0.jjtGetNumChildren() && newParent.jjtGetNumChildren() != 0) {
-                    returnNode = newParent;
+                        return children.get(0);
+                    default:
+                        if (functionMetadata.name().equals(INCLUDE_REGEX)) {
```
Comment (lbschanno) [inline warehouse/query-core/src/main/java/datawave/query/jexl/visitors/RegexFunctionVisitor.java:84]: Not required for approval, but I'd like to see some documentation here explaining why we create an OR node here vs. an AND node for clarity.

File: warehouse/query-core/src/main/java/datawave/query/jexl/visitors/RegexFunctionVisitor.java [hunk 3]
```diff
@@ -47,61 +56,66 @@ public Object visit(ASTFunctionNode node, Object data) {
         FunctionJexlNodeVisitor functionMetadata = new FunctionJexlNodeVisitor();
         node.jjtAccept(functionMetadata, null);
         
-        if (functionMetadata.name().equals("includeRegex") || functionMetadata.name().equals("excludeRegex")) {
+        if (functionMetadata.name().equals(INCLUDE_REGEX) || functionMetadata.name().equals(EXCLUDE_REGEX)) {
```
Comment (lbschanno) [inline warehouse/query-core/src/main/java/datawave/query/jexl/visitors/RegexFunctionVisitor.java:59]: Nice usage of constants here.

**Review by lbschanno — DISMISSED** [review]

File: warehouse/query-core/src/main/java/datawave/query/jexl/visitors/RegexFunctionVisitor.java (hunk 2 above)
Comment (apmoriarty) [reply warehouse/query-core/src/main/java/datawave/query/jexl/visitors/RegexFunctionVisitor.java:84]: Done!

**Review by lbschanno — APPROVED** [review]

**Review by avgAGB — APPROVED** [review]

**Pull request merged**